airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/TestSource.java

```diff
@@ -310,8 +310,18 @@ public void testEmptyStateIncrementalIdenticalToFullRefresh() throws Exception {
   private ConfiguredAirbyteCatalog withSourceDefinedCursors(ConfiguredAirbyteCatalog catalog) {
     ConfiguredAirbyteCatalog clone = Jsons.clone(catalog);
     for (ConfiguredAirbyteStream configuredStream : clone.getStreams()) {
-      if (configuredStream.getSyncMode() == INCREMENTAL && configuredStream.getStream().getSourceDefinedCursor()) {
-        configuredStream.setCursorField(configuredStream.getStream().getDefaultCursorField());
+      if (configuredStream.getStream().getSupportedSyncModes().contains(io.airbyte.protocol.models.SyncMode.INCREMENTAL)) {
+        configuredStream.setSyncMode(io.airbyte.protocol.models.SyncMode.INCREMENTAL);
+        if (Optional.ofNullable(configuredStream.getStream().getSourceDefinedCursor()).orElse(false)) {
+          configuredStream.setCursorField(configuredStream.getStream().getDefaultCursorField());
+        } else {
```

Comment:
as the todo describes, this is really bad. i don't think we need to block postgres on fixing this, but as part of our goal to "launch" incremental, we need to figure out how to include it into our standard tests more ergonomically and transparently. will add an issue for next week. https://github.com/airbytehq/airbyte/issues/1217